ably-iosTests/ARTRealtimeInitTest.m

```diff
@@ -93,10 +94,14 @@ -(void)testInitWithPort {
 
 -(void) testInitWithKey {
     XCTestExpectation *expectation = [self expectationWithDescription:@"testInitWithKey"];
-    [self getBaseOptions:^(ARTClientOptions * options) {
+    [self getBaseOptions:^(ARTClientOptions *options) {
         _realtime = [[ARTRealtime alloc] initWithKey:options.key];
         [_realtime.eventEmitter on:^(ARTRealtimeConnectionState state, ARTErrorInfo *errorInfo) {
-            if(state == ARTRealtimeConnected) {
+            if (state == ARTRealtimeConnecting) {
+                XCTAssertEqual(_realtime.auth.options.key, options.key);
+                // FIXME: Temporary, because the environment is nil instead of `staging` because
+                //instanciating a new RealtimeClient with Key will create a "standard" ClientOptions
+                XCTAssertEqual(((ARTClientOptions *)_realtime.auth.options).environment, nil);
```

Comment:
I think it's OK to test this like that, we don't necessarily need to test that it connects to `sandbox` by default when we are only testing that a single key string works